src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -148,7 +152,17 @@ private bool ReadLineCore(bool consumeKeys)
                     // unprocessed data, or from an actual stdin read.
                     bool previouslyProcessed;
                     ConsoleKeyInfo keyInfo = ReadKey(out previouslyProcessed);
-                    if (!consumeKeys && keyInfo.Key != ConsoleKey.Backspace) // backspace is the only character not written out in the below if/elses.
+                    List<ConsoleKey> denyList = new List<ConsoleKey>
```

Comment:
A new `denyList` instance is being created on each iteration. It would probably make more sense if this was lifted to a field.